### ISSUE TITLE
IntuneAppProtectionPolicyAndroid - AppGroupType & Managedidentity not pulling correctly

### ISSUE THREAD

**Issue by MichealKennard** [microsoft/Microsoft365DSC#2467]
#### Details of the scenario you tried and the problem that is occurring
Performed export of all Intune components against target tenant. Have performed several and each time the fields referenced have to be modified manually before compiling as it doesnt pull correctly.

#### Verbose logs showing the problem
No logs, just screen snip showing issue
![image](https://user-images.githubusercontent.com/116776712/198611814-934ebcc4-fb07-49cf-b4ef-bff963956b99.png)

#### Suggested solution to the issue
Not sure, but potentially adjust logic so it correctly adds "" to variable input or $

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
Windows 10 via Powershell ISE

# insert configuration here
```

#### The operating system the target node is running
<!--
    Please provide as much as possible about the target node, for example
    edition, version, build and language.
    On OS with WMF 5.1 the following command can help get this information.

    Get-ComputerInfo -Property @(
        'OsName',
        'OsOperatingSystemSKU',
        'OSArchitecture',
        'WindowsVersion',
        'WindowsBuildLabEx',
        'OsLanguage',
        'OsMuiLanguages')
-->

#### Version of the DSC module that was used ('dev' if using current dev branch)
Current 1.22.1026.2 but have seen this on previous versions as well testing last couple days.

**Comment by mlhickey** [comment]
Investigating, unclear why True is being generated instead of $True as the output is the same for all resources:
`            $params = @{
                DisplayName           = $policy.displayName
                Ensure                = 'Present'
                Credential            = $Credential
                ApplicationID         = $ApplicationId
                TenantId              = $TenantId
                ApplicationSecret     = $ApplicationSecret
                CertificateThumbprint = $CertificateThumbprint
                ManagedIdentity       = $ManagedIdentity.IsPresent
            }`

Did notice there is an error generating AppGroupType  where the value is not wrapped in quotes.

**Comment by mlhickey** [comment]
Found it - incorrect assignment being created in get-InputParameters:

`AppGroupType                            = @{Type = 'ComplexParameter' ; ExportFileType = 'NA'; };`
`Managedidentity                         = @{Type = 'Credential'       ; ExportFileType = 'NA'; };`

which is incorrect.

**Comment by mlhickey** [comment]
Fix added to PR #2460

**Comment by menswearUK** [comment]
@mlhickey Thanks, I'd missed this issue issue when it was logged and haven't added in a fix for managedidentity yet.  I'll get that added.  I'd spotted there was an issue with AppGroupType and think I've already addressed that but I'll try a few different configs to be sure.
